client/lib/sentry/index.ts

```diff
@@ -159,15 +159,10 @@ export async function initSentry( { beforeSend, userId }: SentryOptions ) {
 			void rejectionQueue.push(
 				// Sentry does this, presumably for browsers being browsers 🤷
 				// https://github.com/getsentry/sentry-javascript/blob/4793df5f515575703d9c83ebf4159ac145478410/packages/browser/src/loader.js#L205
-				// eslint-disable-next-line no-nested-ternary
-				'reason' in exceptionEvent
-					? exceptionEvent.reason
-					: // @ts-expect-error Fixing up browser weirdness
-					'detail' in exceptionEvent && 'reason' in exceptionEvent.detail
-					? // @ts-expect-error Fixing up browser weirdness
-					  exceptionEvent.detail.reason
-					: exceptionEvent
+				// @ts-expect-error Fixing up browser weirdness
+				exceptionEvent?.reason ?? exceptionEvent?.detail?.reason ?? exceptionEvent
```

Comment:
The only way this is safer is when `exceptionEvent` is not an object - then the `in` operator would crash with a `TypeError: Cannot use 'in' operator to search for 'reason'`. It is safer, though, and given the rest of the checks are pretty safe, I don't see any other way for this to break.

Reply:
That makes sense, then! The crash must happen when `exceptionEvent.detail` is truthy, but not an object.